### DISCUSSION TITLE
Don't completely log out when multiple concurrent logins occur

### DISCUSSION THREAD

**Discussion by Mine181** [ppy/osu#27348] [Ideas]
Since Lazer natively supports Mac OS, Linux, Android and IOS, someone could want to play on another device while away from their main device, so having to log in everytime to play on different devices and having to go through email verification is a little tedious.

**Comment by bdach** [comment]
This is not possible due to architectural concerns. The spectator and multiplayer servers are not able to service multiple gameplay sessions from a single user. I do not foresee this changing.

**Reply by ziv-vy** [reply]
Then I think that even a measure as drastic as closing the game on the other device would be a better idea. I do not want to ever log back in on another device, especially with 2fa now being a thing.

**Comment by cihe13375** [comment]
> having to log in everytime to play on different devices and having to go through email verification is a little tedious

At least on stable this is not an issue, since it logins automatically if the credential is saved, and the email verification only happens when logging in on a device for the first time. There is one caveat though: the user should quit the client on device A before logging in on device B, otherwise the two clients would "compete" with each other, leading to an infinite cycle of login/logout (I have encountered this several times).

idk the case of lazer.

**Reply by bdach** [reply]
Verification happens on every fresh login in lazer and that is also explicitly intentional.

**Reply by peppy** [reply]
To clarify, the situation is no worse on lazer when compared to stable, we just fixed the competing-devices part.

Please exit the game on one device before opening on another and you will have zero issues and zero re-verifications.

**Reply by nb-programmer** [reply]
> Verification happens on every fresh login in lazer and that is also explicitly intentional.

This is actually kind of wrong since if I leave my game running on Desktop, then open Osu on Android, it immediately logs out of Desktop. Then when I exit on Android and try to sign in on Desktop, it re-sends verification email.

Like @peppy said we need to fully quit the game on one device then open on another. Idk, maybe the app can keep a flag or a token for re-logging without verification (unless it was manually logged out)

**Reply by peppy** [reply]
End goal will definitely be to have the client NOT lose the login session when this comes up. I see this as pretty high priority because it's really frustrating to get logged out.

A button should appear in the login panel when this occurs (on the client which was disconnected) putting it in an offline state until the button is pressed, or the game is restarted. Either of these actions would disconnect the *other* client and put it in the sme state.
